File: java/RemoveDuplicatesFromSortedArray.java
// Runtime: 1 ms
// Memory Usage: 40.5 MB

class Solution {
    public int removeDuplicates(int[] nums) {
        int index = 0;
        int n = nums.length;

        if (n == 0) {
            return 0;
        }

        for (int i = 0; i < n - 1; i++) {
            if (nums[i] != nums[i + 1]) {
                nums[index++] = nums[i];
            }
        }

        nums[index] = nums[n - 1];
        return index + 1;
    }
}

File: python/RemoveDuplicatesFromSortedArray.py
# Runtime: 107 ms
# Memory Usage: 14.9 MB

class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count = 0

        if len(nums) == 1:
            return 1

        for i in range(len(nums)-1):
            if nums[i] != nums[i+1]:
                nums[count] = nums[i]
                count = count + 1

        nums[count] = nums[len(nums)-1]
        return count + 1
